pkg/sql/sem/builtins/builtins.go

```diff
@@ -2496,11 +2496,42 @@ var regularBuiltins = map[string]builtinDefinition{
 		},
 	),
 
+	"make_date": makeBuiltin(
+		tree.FunctionProperties{Category: builtinconstants.CategoryDateAndTime},
+		tree.Overload{
+			Types:      tree.ParamTypes{{Name: "year", Typ: types.Int}, {Name: "month", Typ: types.Int}, {Name: "day", Typ: types.Int}},
+			ReturnType: tree.FixedReturnType(types.Date),
+			Fn: func(_ context.Context, evalCtx *eval.Context, args tree.Datums) (tree.Datum, error) {
+				year := int(tree.MustBeDInt(args[0]))
+				month := time.Month(int(tree.MustBeDInt(args[1])))
+				day := int(tree.MustBeDInt(args[2]))
+				if year == 0 {
+					return nil, errors.Newf("date field value out of range: %d-%02d-%02d", year, month, day)
```

Comment:
Should this be a `pgerror` with a specific code? (I think there's an invalid parameters one).

Is this a standard error message format? It feels quite terse. I'm not sure I'd understand how to interpret it if I encountered it in the wild.

Reply:
it most likely should! i originally didn't have this error message, but encountered it while testing the behavior out in postgres; so, i mimicked that message🥸. you make a good point - i shall change it to be less terse/more understandable/using pgerror

Reply:
to see which code PG uses, you can add this to your `~/.psqlrc` file:
```
-- Show error codes and location.
\set VERBOSITY verbose
```

Then if you connect to postgres using `psql`, any error it shows will also show you the pgcode